packages/flutter/lib/src/widgets/app.dart

```diff
@@ -1210,7 +1210,9 @@ class WidgetsApp extends StatefulWidget {
     _debugShowWidgetInspectorOverrideNotifier.value = value;
   }
 
-  static final ValueNotifier<bool> _debugShowWidgetInspectorOverrideNotifier = ValueNotifier<bool>(false);
+  static final ValueNotifier<bool> _debugShowWidgetInspectorOverrideNotifier =
+    exemptFromRelease(() => ValueNotifier<bool>(false));
```

Comment:
Is the dart treeshaker/optimizer smart enough to turn this into 

```suggestion
  static final ValueNotifier<bool> _debugShowWidgetInspectorOverrideNotifier = ValueNotifier<bool>(false);
```

when `kFlutterMemoryAllocationsEnabled` is false? Or do we still pay for the method calls and instantiating the closure?

Reply:
I am not sure how to test if `() => ValueNotifier<bool>(false))()` is converted to `ValueNotifier<bool>(false)`. Assuming it is not these are options:

1. Go with kernel transformations
2. Wait for macros and for now have tech debt`kFlutterMemoryAllocationsEnabled ? ValueNotifier<bool>(false) : exemptFromRelease(() => ValueNotifier<bool>(false));
3. Agree that one closure per app run is acceptable cost